.build-tools/builtin-authentication-profiles.yaml

```diff
@@ -103,3 +103,70 @@ azuread:
           - AzurePublicCloud
           - AzureChinaCloud
           - AzureUSGovernmentCloud
+
+gcp:
+  - title: "GCP API Authentication with Service Account Key"
+    description: |
+      Authenticate authenticates API calls with the given service account or refresh token JSON credentials.
+    metadata:
+      - name: privateKeyID
+        required: true
+        sensitive: true
+        description: |
+          The GCP private key id. Replace with the value of "private_key_id" field of the Service Account Key file.
+        example: '"privateKeyID"'
+      - name: privateKey
+        required: true
+        sensitive: true
+        description: |
+          The GCP credentials private key. Replace with the value of "private_key" field of the Service Account Key file.
+        example: '"-----BEGIN PRIVATE KEY-----\nMIIEvgIBADAsUw+2/y/MfhhkIKQTgnoiIZFHximeQN67VF9pifNR/D0k\\n-----END PRIVATE KEY-----\n"'
+      - name: type
+        type: string
+        required: true
+        description: |
+          The GCP credentials type.
+        example: '"service_account"'
+      - name: projectID
+        type: string
+        required: true
+        description: |
+          GCP project id.
+        example: '"projectID"'
+      - name: clientEmail
+        type: string
+        required: true
+        description: |
+          GCP client email.
+        example: '"client@email.com"'
+      - name: clientID
+        type: string
+        required: true
+        description: |
+          The GCP client ID.
+        example: '"0123456789-0123456789"'
+      - name: authURI
```

Comment:
All these fields till the end: "authURI", "tokenURI", "authProviderX509CertURL"... aren't they supposed to come from the OpenID Configuration document? https://accounts.google.com/.well-known/openid-configuration Normally for OAuth2 users just need to provide the endpoint (in this case `http://accounts.google.com`), then the caller knows to add `well-known/openid-configuration` (it's "well-known" after all) to fetch the information below.

Reply:
I used the following document for the description of those fields: https://googleapis.github.io/google-api-python-client/docs/client-secrets.html